src/org/thoughtcrime/securesms/NewConversationActivity.java

```diff
@@ -99,18 +99,21 @@ public void onContactSelected(int specialId, String addr) {
     else {
       int contactId = dcContext.lookupContactIdByAddr(addr);
       if (contactId!=0 && dcContext.getChatIdByContactId(contactId)!=0) {
-        openConversation(dcContext.createChatByContactId(contactId));
+        openConversation(dcContext.getChatIdByContactId(contactId));
```

Comment:
why is this needed? the create-chat thing was used to unblock chats or something like that IIRC :thinking: 

in any case, what is the downside of using createChatByContactId? (just want to know)

Reply:
in a perfect world, both would be fine, however, `getChatIdByContactId()` is a read-only-operation - and therefore a precaution against bugs as the weirdly changed name